src/ck-core/ck.C

```diff
@@ -2449,10 +2447,12 @@ void CkGroupExtSend_multi(int gid, int pe, int epIdx, int num_bufs, char **bufs,
   CkGroupID gId;
   gId.idx = gid;
 
-  if (pe == -1)
```

Comment:
If most callers of this function have arguments known at compile time that determine which of these three branches they will take, it might be nice to have an API that does not have a runtime branch on compile time information. This would mean `CkGroupExtSend`, `CkGroupExtSendMulti`, `CkGroupExtBroadcast` functions. They could call a static function that does the first half of `CkGroupExtSend_multi` to avoid code duplication. This is very detail-oriented though and not terribly important.

Reply:
Yeah, I was considering doing something like this but fine grained performance testing hasn't shown any difference due to the branching, and I'm not sure how much code duplication it would result in (there are two variants of CkGroupExtSend right now which don't really share the first half of the code). My idea is to do this in the future if it does turn into a bottleneck.

Reply:
Also, the branch is necessary in any case because a group proxy could be for a broadcast or an individual element and the check is done at runtime. What is not necessary is to add `CkSendMsgBranchMulti` to the branch (it could have its own separate function), but like I said I haven't seen any performance regression.